File: Algorithm/Python/944. Delete Columns to Make Sorted.py
# -*- coding: utf-8 -*-
# Created on Fri Nov 23 2018 20:50:28
# Author: WuLC
# EMail: liangchaowu5@gmail.com

class Solution(object):
    def minDeletionSize(self, A):
        """
        :type A: List[str]
        :rtype: int
        """
        m, n = len(A), len(A[0])
        count = 0
        for i in xrange(n):
            for j in xrange(1, m):
                if A[j][i] < A[j-1][i]:
                    count += 1
                    break
        return count

File: Algorithm/C++/944. Delete Columns to Make Sorted.cc
/*
 * Created on Fri Nov 23 2018 20:56:19
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

class Solution {
 public:
  int minDeletionSize(vector<string>& A) {
    int m = A.size(), n = A[0].size();
    int count = 0;
    for (int i = 0; i < n; i++) {
      for (int j = 1; j < m; j++) {
        if (A[j][i] < A[j-1][i]) {
          count++;
          break;
        }
      }
    }
    return count;
  }
};